core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -558,10 +558,12 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
         syncPersistDirectory((InodeDirectory) inode, journalContext);
       }
     }
-    if (pathIndex < (pathComponents.length - 1) || currentInodeDirectory.getChild(name) == null) {
+    if ((pathIndex < (pathComponents.length - 1) || currentInodeDirectory.getChild(name) == null)
+        && options.getOperationTimeMs() > currentInodeDirectory.getLastModificationTimeMs()) {
       // (1) There are components in parent paths that need to be created. Or
       // (2) The last component of the path needs to be created.
-      // In these two cases, the last traversed Inode will be modified.
+      // In these two cases, the last traversed Inode will be modified if the new timestamp is after
```

Comment:
Why is this association important? I thought the `options.getOperationTimeMs()` is the UFS timestamp of the final component. Why is there a relationship between the final UFS component, and the timestamp of the last existing Alluxio inode?

For example:
Alluxio inode: `/a`
UFS path: `/a/b/c/file`

What is the relationship between the timestamps of those two entities?

Reply:
The original idea I believe is to update the last modified time of parent directory when a file or directory is created in them. I added an additional condition so that for UFS timestamp loaded to be earlier than current modification time of the parent should not cause the parent last modification time to be changed backwards.

Reply:
I see. This means that if we are loading `/a/b/c/file` (with old timestamp) but only alluxio inode `/a` exists, it will not update the last modified time of `/a`.